File: frontend/src/app/services/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Comment } from '../models/plan.model';

@Injectable({
  providedIn: 'root'
})
export class CommentService {
  private apiUrl = 'http://localhost:8080/api';

  constructor(private http: HttpClient) {}

  createComment(initiativeId: number, comment: Comment): Observable<Comment> {
    return this.http.post<Comment>(`${this.apiUrl}/initiatives/${initiativeId}/comments`, comment);
  }

  getComments(initiativeId: number): Observable<Comment[]> {
    return this.http.get<Comment[]>(`${this.apiUrl}/initiatives/${initiativeId}/comments`);
  }

  updateComment(commentId: number, comment: Comment): Observable<Comment> {
    return this.http.put<Comment>(`${this.apiUrl}/comments/${commentId}`, comment);
  }

  deleteComment(commentId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/comments/${commentId}`);
  }
}

File: backend/main-service/src/main/java/com/plantrack/backend/service/CommentService.java
package com.plantrack.backend.service;

import com.plantrack.backend.model.Comment;

import java.util.List;

public interface CommentService {

    public Comment createComment(Long initiativeId, Comment comment);
    public List<Comment> getCommentsByInitiative(Long initiativeId);
    public Comment updateComment(Long commentId, Comment updatedComment);
    public void deleteComment(Long commentId);

}
